### PR TITLE
Add inclusion/exclusion support on bulksetampl.

### PR DESCRIPTION
**Pull request by MichaelBroughton**
Adds the `exclude` flag to BulkSetAmpl to provide greater flexibility for bulk writing. Would be helpful on https://github.com/tensorflow/quantum/pull/458

### CODE REVIEW

**Review by sergeisakov — APPROVED** [review]

**Comment by sergeisakov** [comment]
LGTM

File: tests/statespace_testfixture.h [hunk 1]
```diff
@@ -820,7 +820,7 @@ void TestBulkSetAmplitude() {
   for(int i = 0; i < 8; i++) {
     state_space.SetAmpl(state, i, 1, 1);
   }
-  state_space.BulkSetAmpl(state, 1, 0, 0, 0);
+  state_space.BulkSetAmpl(state, 1, 0, 0, 0, false);
```
Comment (95-martin-orion) [inline tests/statespace_testfixture.h:823]: Since `false` is the default, we should have at least one test case where this argument is omitted.

**Review by 95-martin-orion — CHANGES_REQUESTED** [review]
One nit on the tests, but the rest looks alright to me.

File: tests/statespace_testfixture.h (hunk 1 above)
Comment (MichaelBroughton) [reply tests/statespace_testfixture.h:823]: Done.

**Review by 95-martin-orion — APPROVED** [review]

**Pull request merged**